core/src/main/java/hudson/model/DirectoryBrowserSupport.java

```diff
@@ -355,7 +366,7 @@ private static String createBackRef(int times) {
     private static void zip(OutputStream outputStream, VirtualFile dir, String glob) throws IOException {
         try (ZipOutputStream zos = new ZipOutputStream(outputStream)) {
             zos.setEncoding(System.getProperty("file.encoding")); // TODO JENKINS-20663 make this overridable via query parameter
-            for (String n : dir.list(glob.length() == 0 ? "**" : glob)) {
+            for (String n : dir.list(glob.isEmpty() ? "**" : glob, null, /* TODO what is the user expectation? */true)) {
```

Comment:
Retaining default excludes functionality here to maintain semantics but I am unsure whether users would want this or not.